src/Rings/mpoly-localizations.jl

```diff
@@ -123,6 +123,11 @@ function rand(S::MPolyPowersOfElement, v1::UnitRange{Int}, v2::UnitRange{Int}, v
   return prod(f^rand(0:9) for f in denominators(S); init = one(R))::elem_type(R)
 end
 
+function rand(rng::Random.AbstractRNG, S::MPolyPowersOfElement, v1::UnitRange{Int}, v2::UnitRange{Int}, v3::UnitRange{Int})
+  R = ambient_ring(S)
+  return prod(f^rand(rng, 0:9) for f in denominators(S); init = one(R))::elem_type(R)
```

Comment:
What I see here seems to contradict my previous comment, really. You are generating the exponents with the same 'seeded' `rng`, right? Then, I do not know what causes the differences in timings.